hikari/impl/interaction_server.py

```diff
@@ -86,26 +86,29 @@
 _USER_AGENT_KEY: typing.Final[str] = "User-Agent"
 _APPLICATION_OCTET_STREAM: typing.Final[str] = "application/octet-stream"
 _JSON_CONTENT_TYPE: typing.Final[str] = "application/json"
-_JSON_TYPE_WITH_CHARSET: typing.Final[str] = f"{_JSON_CONTENT_TYPE}; charset={_UTF_8_CHARSET}"
 _TEXT_CONTENT_TYPE: typing.Final[str] = "text/plain"
-_TEXT_TYPE_WITH_CHARSET: typing.Final[str] = f"{_TEXT_CONTENT_TYPE}; charset={_UTF_8_CHARSET}"
 
 
 class _Response:
-    __slots__: typing.Sequence[str] = ("_content_type", "_files", "_payload", "_status_code")
+    __slots__: typing.Sequence[str] = ("_content_type", "_charset", "_files", "_payload", "_status_code")
 
     def __init__(
         self,
         status_code: int,
         payload: typing.Optional[bytes] = None,
         *,
         content_type: typing.Optional[str] = None,
+        charset: typing.Optional[str] = None,
         files: typing.Sequence[files_.Resource[files_.AsyncReader]] = (),
     ) -> None:
-        if payload and not content_type:
-            content_type = _TEXT_TYPE_WITH_CHARSET
+        if payload:
+            if not content_type:
+                content_type = _TEXT_CONTENT_TYPE
+            if not charset:
+                charset = _UTF_8_CHARSET
```

Comment:
In most cases this will be `_UTF_8_CHARSET`, should we just default to it instead of allowing different values to it?

Reply:
We want to add charset only if the payload is not None, just like content_type

Reply:
If we want to keep things abstract, this is fine. Ignore my other idea :)